pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -124,16 +139,19 @@ func newRepositoryWithClient(
 		return nil, fmt.Errorf("%s is required", DockerRegistryURLEnvVar)
 	}
 
-	pullthrough := false
-	if value, ok := options["pullthrough"]; ok {
-		if b, ok := value.(bool); ok {
-			pullthrough = b
-		}
+	pullthrough := getBoolOption("pullthrough", false, options)
+
+	var acceptschema2 bool
+
+	if os.Getenv(EnforceAcceptSchema2EnvVar) == "" {
```

Comment:
There's a helper [function](https://github.com/openshift/origin/blob/master/pkg/dockerregistry/server/quotarestrictedblobstore.go#L36) to process these env vars. The refactor could be done as a follow-up though.